### PR TITLE
[WIP] Starting restructuring into CI Folder

### PR DESCRIPTION
**Pull request by gvollbach**
Since I think there are more tools to come, in my opinion it would make sense, to have a separate folder for the CI scripts.

- We do not pollute the root directory of ILIAS anymore with scripts or config files.
- We have a simple entry point, if devs would like to view the scripts
  -  If we will have a building process for ILIAS releases in the future, it will be really simple to remove all CI scripts.
- The only "script" which will stay in the root directory will be the .travis.yml file

### CODE REVIEW

File: CI/PHP-CS-Fixer/code-format.php_cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<?php
+
+require_once __DIR__ . '/libs/composer/vendor/autoload.php';
+
+$finder = PhpCsFixer\Finder::create()
+	->in(array(
+		__DIR__ . '/Services',
+		__DIR__ . '/Modules',
+		__DIR__ . '/src',
+		__DIR__ . '/tests'
```
Comment (mjansenDatabay) [inline CI/PHP-CS-Fixer/code-format.php_cs]: Maybe we should include `setup` here as well, except the `sql` folder.

Comment (mjansenDatabay) [reply CI/PHP-CS-Fixer/code-format.php_cs]: And `/cron`, `/include`

**Review by alex40724 — APPROVED** [review]
Thanks!

File: CI/PHP-CS-Fixer/code-format.php_cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+		__DIR__ .  '/../../include',
+		__DIR__ .  '/../../Modules',
+		__DIR__ .  '/../../Services',
+		__DIR__ .  '/../../setup',
+		__DIR__ .  '/../../src',
+		__DIR__ .  '/../../tests'
+	))
+;
+
+return PhpCsFixer\Config::create()
```
Comment (legionth) [inline CI/PHP-CS-Fixer/code-format.php_cs:18]: This looks very similar to the code formatter I've been testing for the GitHooks.
I found some additional rules that may be should applied, so our coding style matches:
```php
return PhpCsFixer\Config::create()
	->setRules([
		'@PSR2' => true,
		'strict_param' => false,
		'cast_spaces' => true,
		'concat_space' => ['spacing' => 'one'],
		'binary_operator_spaces' => true,
		'unary_operator_spaces' => true,
		'function_typehint_space' => true,
		'return_type_declaration' => ['space_before' => 'one'],
		'whitespace_after_comma_in_array' => true,
	])
	->setFinder($finder);
```
I tried this with the coding style examples provided by #1838 and it worked very well. Maybe you could verify this or we can check this together :+1:

File: CI/PHP-CS-Fixer/code-format.php_cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+	->setRules([
+        '@PSR2'                   => true,
+        'strict_param'            => false,
+        'array_syntax'            => ['syntax' => 'short'],
```
Comment (legionth) [inline CI/PHP-CS-Fixer/code-format.php_cs]: Is this rule `array_syntax` currently a must or should we leave it out?

File: CI/PHP-CS-Fixer/code-format.php_cs (hunk 2 above)
Comment (gvollbach) [reply CI/PHP-CS-Fixer/code-format.php_cs:18]: @legionth Thx, I've added the config :+1:

**Comment by Amstutz** [comment]
According to MJ in the todays TB meeting, this seems to be a draft an not yet completed. Please close and reopen as "draft".

**Pull request closed without merging**